.github/stale.yml

```diff
@@ -0,0 +1,18 @@
+# Number of days of inactivity before an issue becomes stale
+daysUntilStale: 30
+# Number of days of inactivity before a stale issue is closed
+daysUntilClose: 7
+# Issues with these labels will never be considered stale
+exemptLabels:
+  - important
+# Label to use when marking an issue as stale
+staleLabel: stale
+# Comment to post when marking an issue as stale. Set to `false` to disable
+markComment: >
+  This issue has been automatically marked as stale because it has not had
+  recent activity. It will be closed if no further activity occurs.
+  If this is still an issue, please make sure it is up to date and if so,
+  add a comment that this is still an issue to keep it open.
+  Thank you for your contributions.
+# Comment to post when closing a stale issue. Set to `false` to disable
+closeComment: false
```

Comment:
These things also apply to `PR`s? maybe `important` label is not sufficient for exempt labels,  how about `crash`、`OOM`、`discussion`、`feature`?

Reply:
It applies to PRs as well.
From by POV and past experience with the bot, I would like to leave it as open as possible. Basically, just get a 30 days reminder on each issue/PR that was not updated. Then, if we just do anything there, the Stale label goes away.
I think `important` label is sufficient, just to have one that is overlooked. If we add more labels, we will just let issues pile up.

Reply:
If we also want to apply to `PR`, `markComment` section should adjust a bit? seems it says only `Issue`, not `PR`.
We can see `Issues` currently opened, some are labeled `crash`, `OOM`, seems cannot fix ASAP. So for these situations, seems we need to also add `important` label as well.

Reply:
I don't think we need. Let the bot comment and we can revisit. Better this way, otherwise we just forget about them and never resolve them.

Reply:
One question: I think for issue, it's easy for our contributor to determine whether it's `important` or not. And can be clearly tagged. So the stale config is correct.

However, for PRs, sometime we can not immediatelly determine whether it's important or not. Because it's base on implementation but not `what this PR solved`. And people who create PR (normally the first time commiter), have no right to tag their PR with any tag. So I think this may sometimes make it harder for new commiter to give us their idea. (I don't want to effect someone's heart about open source contribution).

Can stale have different config namespace between PR or issue ? For example, for PR, the time stay longer. Or for PR, using more aspects to keep it not stale (for example, if some PR's milestone is `future`, we don't stale it)

Reply:
I don't think we can split the configs. The one thing I can do is just ignore PRs. Even though I think it's important not to let PRs open for so long and the bot could help us push on them.
Basically, I think once an issue or pr is open, we need to push on that for an action plan and a resolution. 30 days should be sufficient to do it. If it's something that takes more time, we will just make a change and have the bot ignore it for another 30 days.

Reply:
We can, see [stale](https://github.com/probot/stale).

Reply:
I didn't know about that @zhongwuzw  :)

So should I add
```
pulls:
  daysUntilStale: 60
  markComment: >
    This pull request has been automatically marked as stale because it has not had
     recent activity. It will be closed if no further activity occurs.
     If this is still applicable, please make sure it is up to date and if so,
     add a comment that this is still an issue to keep it open.
     Thank you for your contributions.
```
?

Reply:
Looks like they have namespace...Options without any `pulls` or `issues ` as common configuration. You can then add `pulls` to override the common configuration and specify it for PR. I can have a try on [SDWebImagePhotosPlugin](https://github.com/SDWebImage/SDWebImagePhotosPlugin)...

Reply:
I added the pulls stale to 60 days and custom message. Let's merge.

Reply:
Seems a solution. Actually what I want maybe like this. However, I read their doc seems this feature is in development and not available.

```
pulls:
    exemptMilestones:
        - future
```

Currently it's OK to merge. 60 days is enough for our contributor to talk about on PR.